doc/style-guide-testing.md

```diff
@@ -0,0 +1,467 @@
+                    allowNoValueEvent = true;
+                    eventsEmitted = 0;
+                    $parent.on('no value', function () {
+                        eventsEmitted += 1;
+                    });
+                    compile('<div my-custom-directive></div>');
+                });
+                it('emits "no value"', function () {
+                    expect(eventsEmitted).toBe(1);
+                });
+            });
+        });
+        describe('update event', function () {
+            var eventPassed, $scope;
+
+            beforeEach(function () {
+                $scope = compile('<div my-custom-directive></div>');
+                eventPassed = false;
+            });
+            it('allows the number by default', function () {
+                $parent.$broadcast('update');
+                expect(eventPassed).toBe(true);
+            });
+            it('allows the number with even true', function () {
+                $scope.even = true;
+                $parent.$broadcast('update');
+                expect(eventPassed).toBe(true);
+            });
+            it('disallows the number with even false', function () {
+                $scope.even = false;
+                $parent.$broadcast('update');
+                expect(eventPassed).toBe(false);
+            });
+        });
+        describe('dirty event', function () {
+            var $child, eventPassed, $scope;
+
+            beforeEach(function () {
+                // Test events coming up from a lower scope
+                $scope = compile('<div my-custom-directive></div>');
+                $parent.on('dirty', function () {
+                    eventPassed = true;
+                });
+            });
+            it('allows the event by default', function () {
+                $child.emit('dirty');
+                expect(eventPassed).toBe(true);
+            });
+            it('allows the event when number is 6', function () {
+                $scope.number = 6;
+                $child.emit('dirty');
+                expect(eventPassed).toBe(true);
+            });
+            it('disallows the event when number is 7', function () {
+                $scope.number = 7;
+                $child.emit('dirty');
+                expect(eventPassed).toBe(false);
+            });
+        });
+    });
+
+That sure is a lengthy example, but carefully notice how anyone could add additional assertions, test cases, alternate setups and different scenarios.  The event testing code is clearly separate from the initialization code.  Promises are used for resolving values.  Events are sent in the correct way through the directive and are tested at the appropriate levels.  Data is flowing through the directive and still you don't tie yourself to any other services or code outside of your directive.
+
+
+Data Sets
+---------
+
+A filter is often tested by providing a series of inputs and asserting the output is correct.
+
+```
```

Comment:
Why are you using a code block here when you don't other places and specifically say in your guid for markdown to not use code blocks but to indent?